libraries/AP_SensorHead/AP_SensorHead.cpp

```diff
@@ -0,0 +1,91 @@
+#include "AP_SensorHead.h"
+
+extern const AP_HAL::HAL& hal;
+
+using namespace SensorHead;
+
+AP_SensorHead AP_SensorHead::_instance{};
+bool AP_SensorHead::_initialized = false;
+
+AP_SensorHead *AP_SensorHead::init()
+{
+    _initialized = true;
+    return &_instance;
+}
+
+AP_SensorHead::AP_SensorHead()
+{
+    // TODO: Segfault for some reason.
```

Comment:
it will segfault as hal.util hasn't been set yet. It's a constructor ordering issue. You'll need to setup these in an init() function, otherwise it will work on some platforms but not others, depending on constructor ordering